### PR TITLE
SDCA

### PR DESCRIPTION
**Pull request by lvilnis**
Add implementation of stochastic dual coordinate ascent solver for classification. Also a few additions to optimizable objectives.

### CODE REVIEW

File: src/main/scala/cc/factorie/optimize/SDCA.scala [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    }
+
+    println("Loaded " + trainInstances.size + " train instances.")
+
+    val sdca = new SDCA(trainInstances.map(_._1), trainInstances.map(t => if (t._2 == 0) -1 else 1), UnivariateDualUpdate.smoothHingeBinaryDualUpdate(), evaluate = Some(evaluate))
+
+    sdca.learn()
+  }
+
+  object SVMLightHelpers {
```
Comment (strubell) [inline src/main/scala/cc/factorie/optimize/SDCA.scala]: Maybe this should be moved into load?

**Comment by strubell** [comment]
otherwise lgtm

**Comment by strubell** [comment]
@lvilnis looking over old PRs now since I think there will be a release soon. Any chance you will move `SVMLightHelpers` into `load` or some similarly appropriate package? :)

**Comment by lvilnis** [comment]
I broke everything so im gonna make a new PR

**Pull request closed without merging**